### PR TITLE
Feature: Improved Plugin Docs

### PR DESCRIPTION
**Pull request by ngandrass**
This PR migrates the existing documentation from the `README.md` file to a mkdocs based web page. It also extends the existing docs and automates the build and deployment process.

Corresponding issue: #39

### CODE REVIEW

File: docs/configuration/initialconfig/automatic.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+webservice for the archive worker, can be done automatically.
+
+The easiest way is to use the automatic configuration feature provided via the
+Moodle admin interface but a fully automated configuration via CLI is also
+supported.
+
+
+## Using the Moodle Admin Interface
+
+!!! info
```
Comment (melanietreitinger) [inline docs/configuration/initialconfig/automatic.md:13]: I am confused: Why would you recommend this for "most users"?
As you are writing in the paragraph above, it is the easiest way to use the automatic configuration?

File: docs/configuration/initialconfig/automatic.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    This is the recommended way to configure the Quiz Archiver Moodle plugin for
+    most users.
+
+1. Navigate to _Site Administration_ > _Plugins_ (1) > _Activity modules_ >
+   _Quiz_ > _Quiz Archiver_ (2)
+2. Click the _Automatic configuration_ button (3)
+3. Enter the URL under which the quit archive worker can be reached (4)
```
Comment (melanietreitinger) [inline docs/configuration/initialconfig/automatic.md:20]: typo: quiz

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+[![GitHub Issues](https://img.shields.io/github/issues/ngandrass/moodle-quiz_archiver)](https://github.com/ngandrass/moodle-quiz_archiver/issues)
+[![GitHub Pull Requests](https://img.shields.io/github/issues-pr/ngandrass/moodle-quiz_archiver)](https://github.com/ngandrass/moodle-quiz_archiver/pulls)
+[![Donate with PayPal](https://img.shields.io/badge/PayPal-donate-orange)](https://www.paypal.me/ngandrass)
+[![Sponsor with GitHub](https://img.shields.io/badge/GitHub-sponsor-orange)](https://github.com/sponsors/ngandrass)
+[![GitHub Stars](https://img.shields.io/github/stars/ngandrass/moodle-quiz_archiver?style=social)](https://github.com/ngandrass/moodle-quiz_archiver/stargazers)
+[![GitHub Forks](https://img.shields.io/github/forks/ngandrass/moodle-quiz_archiver?style=social)](https://github.com/ngandrass/moodle-quiz_archiver/network/members)
+[![GitHub Contributors](https://img.shields.io/github/contributors/ngandrass/moodle-quiz_archiver?style=social)](https://github.com/ngandrass/moodle-quiz_archiver/graphs/contributors)
+
+Archives quiz attempts as PDF and HTML files for long-term storage independent
```
Comment (melanietreitinger) [inline docs/index.md:14]: Add at the beginning of the sentence: "This plugin creates"

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+&nbsp; &nbsp; &nbsp;
+[:material-cog: Configuration](/configuration){ .md-button }
+&nbsp; &nbsp; &nbsp;
+[:material-account: Usage](/usage){ .md-button }
+
+
+## Features
+
+- Archiving of quiz attempts as PDF and HTML files
+- Support for file submissions / attachments (e.g., essay files)
```
Comment (melanietreitinger) [inline docs/index.md:52]: remove comma after "e.g."

File: docs/configuration/initialconfig/automatic.md (hunk 2 above)
Comment (ngandrass) [reply docs/configuration/initialconfig/automatic.md]: Fixed with c4653c4

File: docs/index.md (hunk 3 above)
Comment (ngandrass) [reply docs/index.md]: Added with aca6b8e

File: docs/configuration/initialconfig/manual.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+![Screenshot: Configuration - Assign Role 1](/assets/configuration/configuration_assign_role_1.png){ .img-thumbnail }
+![Screenshot: Configuration - Assign Role 2](/assets/configuration/configuration_assign_role_2.png){ .img-thumbnail }
+![Screenshot: Configuration - Assign Role 3](/assets/configuration/configuration_assign_role_3.png){ .img-thumbnail }
+
+
+## Setup Webservice
+
+The quiz archive worker service interacts with the Moodle platform using the
+Moodle webservice API. Therefore, it must be enabled and a corespongind external
```
Comment (melanietreitinger) [inline docs/configuration/initialconfig/manual.md:65]: Typo in "corespongind"

File: docs/configuration/initialconfig/pitfalls.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+## The archive worker service cannot access the Moodle instance
+
+If the archive worker service can not access your Moodle instance, you should
+check the following:
+
+1. Ensure that the archive worker service is able to resolve the hostname of your
+   Moodle instance.
+    - You can test this by running `nslookup yourmoodle.tld` on the machine that
+    runs the archive worker service.
+    - If the lookup fails, check the DNS settings on the machine thar runs the
```
Comment (melanietreitinger) [inline docs/configuration/initialconfig/pitfalls.md:27]: Typo in "thar"

**Review by melanietreitinger — COMMENTED** [review]
The documentation is very elaborate, coherent and well structured!
Not much to add from my side, except some very few typos! 👏

File: docs/configuration/initialconfig/manual.md (hunk 5 above)
Comment (ngandrass) [reply docs/configuration/initialconfig/manual.md]: Fixed with b76e2df

File: docs/configuration/initialconfig/pitfalls.md (hunk 6 above)
Comment (ngandrass) [reply docs/configuration/initialconfig/pitfalls.md]: Fixed with b76e2df

**Comment by ngandrass** [comment]
Thanks a LOT @melanietreitinger for going through the docs :clap: 

I'll prepare everything for the merge and will release it all together after my Moodle 4.5 (LTS) tests are complete. That should be within the next days and no later than next week :)

**Pull request merged**